lib/elastic_apm/trace_context.rb

```diff
@@ -71,12 +64,36 @@ def hex_flags
       format('%02x', flags.to_i(2))
     end
 
+    def ensure_parent_id
+      @parent_id ||= hex_id(ID_N)
+      @parent_id
+    end
+
     def child
-      dup.tap { |tc| tc.span_id = SecureRandom.hex(8) }
+      dup.tap do |tc|
+        tc.parent_id = tc.id
+        tc.id = hex_id(ID_N)
+      end
     end
 
     def to_header
-      format('%s-%s-%s-%s', version, trace_id, span_id, hex_flags)
+      format('%s-%s-%s-%s', version, trace_id, id, hex_flags)
+    end
+
+    # @deprecated Use parent_id instead
+    def span_id
+      @parent_id
+    end
+
+    # @deprecated Use parent_id instead
+    def span_id=(span_id)
+      @parent_id = span_id
+    end
+
+    private
+
+    def hex_id(len)
```

Comment:
Your time in Go is making its mark on your variable names – saving 3 characters here 😀